pkg/operator/controller/controller.go

```diff
@@ -251,8 +252,32 @@ func (r *reconciler) ensureRouterNamespace() error {
 // ensureRouterForIngress ensures all necessary router resources exist for a
 // given clusteringress.
 func (r *reconciler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress) error {
-	expected, err := r.ManifestFactory.RouterDeployment(ci)
+	statsSecret, err := r.ManifestFactory.RouterStatsSecret(ci)
 	if err != nil {
+		return fmt.Errorf("failed to build router stats secret: %v", err)
+	}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: statsSecret.Namespace, Name: statsSecret.Name}, statsSecret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to get router stats secret %s/%s, %v", statsSecret.Namespace, statsSecret.Name, err)
+		}
+
+		err = r.Client.Create(context.TODO(), statsSecret)
+		if err == nil {
+			logrus.Infof("created router stats secret %s/%s", statsSecret.Namespace, statsSecret.Name)
+		} else if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create router stats secret %s/%s: %v", statsSecret.Namespace, statsSecret.Name, err)
+		}
+	}
+
+	expected, err := r.ManifestFactory.RouterDeployment(ci, r.EnableRouterMetrics)
+	if err != nil {
+		r.Client.Delete(context.TODO(), statsSecret)
```

Comment:
Yeah, don't worry about this for now — we have a general bug for cleanup and part of that will be refactoring to use something like the SA as the anchor for siblings of the deployment.